src/encdatavault_fmt_plug.c

```diff
@@ -26,24 +28,32 @@ john_register_one(&fmt_encdadatavault);
 #include "formats.h"
 #include "pbkdf2_hmac_sha256.h"
 
-#define FORMAT_TAG              "$encdv$"
-#define FORMAT_TAG_LENGTH       (sizeof(FORMAT_TAG) - 1)
-#define FORMAT_LABEL            "ENCDataVault"
-#define FORMAT_NAME             ""
-#define ALGORITHM_NAME          "MD5 / PBKDF2-SHA256, AES"
-#define BENCHMARK_COMMENT       ""
-#define BENCHMARK_LENGTH        0x107
-#define PLAINTEXT_LENGTH        125
-#define BINARY_SIZE             0
-#define BINARY_ALIGN            1
-#define SALT_SIZE               sizeof(struct salt)
-#define SALT_ALIGN              4
+#define FORMAT_TAG_MD5            "$encdv$"
+#define FORMAT_TAG_PBKDF2         "$encdv-pbkdf2$"
+#define FORMAT_TAG_MD5_LENGTH     (sizeof(FORMAT_TAG_MD5) - 1)
+#define FORMAT_TAG_PBKDF2_LENGTH  (sizeof(FORMAT_TAG_PBKDF2) - 1)
+#define FORMAT_LABEL_MD5          "ENCDataVault-MD5"
+#define FORMAT_LABEL_PBKDF2       "ENCDataVault-PBKDF2"
+#define FORMAT_NAME               ""
+#define ALGORITHM_NAME_MD5        "MD5, AES"
 #ifdef SIMD_COEF_32
-#define MIN_KEYS_PER_CRYPT      SSE_GROUP_SZ_SHA256
-#define MAX_KEYS_PER_CRYPT      SSE_GROUP_SZ_SHA256
+#define ALGORITHM_NAME_PBKDF2     "PBKDF2-SHA256 HMAC-256 " SHA256_ALGORITHM_NAME
 #else
-#define MIN_KEYS_PER_CRYPT      1
-#define MAX_KEYS_PER_CRYPT      4
+#define ALGORITHM_NAME_PBKDF2     "PBKDF2-SHA256 HMAC-256 32/" ARCH_BITS_STR
+#endif
+#define BENCHMARK_COMMENT         ""
+#define BENCHMARK_LENGTH          0x107
+#define PLAINTEXT_LENGTH          125
+#define BINARY_SIZE               0
+#define BINARY_ALIGN              1
+#define SALT_SIZE                 sizeof(struct salt)
+#define SALT_ALIGN                4
+#ifdef SIMD_COEF_32
+#define MIN_KEYS_PER_CRYPT        SSE_GROUP_SZ_SHA256
+#define MAX_KEYS_PER_CRYPT        SSE_GROUP_SZ_SHA256
+#else
+#define MIN_KEYS_PER_CRYPT        1
+#define MAX_KEYS_PER_CRYPT        1
 #endif
```

Comment:
Part of the reason to split into two formats is not to force `*_KEYS_PER_CRYPT` to `SSE_GROUP_SZ_SHA256` in the MD5-based format (and once we implement SIMD for it later, to use the constants proper for the SIMD implementation of MD5, not SHA-256). So you need a separate set of MIN/MAX per format.